protocol/messenger_communities.go

```diff
@@ -1095,6 +1172,223 @@ func (m *Messenger) handleSyncCommunity(messageState *ReceivedMessageState, sync
+				// No prior messages to be archived, so we just kick off the archive creation loop
+				// for future archives
+				go m.communitiesManager.StartHistoryArchiveTasksInterval(c, messageArchiveInterval)
+			} else if durationSinceLastArchive < messageArchiveInterval {
+				// Last archive is less than `interval` old, wait until `interval` is complete,
+				// then create archive and kick off archive creation loop for future archives
+				// Seed current archive in the meantime
+				err := m.communitiesManager.SeedHistoryArchiveTorrent(c.ID())
+				if err != nil {
+					m.logger.Debug("failed to seed history archive", zap.Error(err))
+				}
+				timeToNextInterval := messageArchiveInterval - durationSinceLastArchive
+
+				log.Println("Starting history archive tasks interval in: ", timeToNextInterval)
+				time.AfterFunc(timeToNextInterval, func() {
+					err := m.communitiesManager.CreateAndSeedHistoryArchive(c.ID(), topics, lastArchiveEndDate, to.Add(timeToNextInterval), messageArchiveInterval)
+					if err != nil {
+						m.logger.Debug("failed to get create and seed history archive", zap.Error(err))
+					}
+					go m.communitiesManager.StartHistoryArchiveTasksInterval(c, messageArchiveInterval)
+				})
+			} else {
+				// Looks like the last archive was generated more than `interval`
+				// ago, so lets create a new archive now and then schedule the archive
+				// creation loop
+				err := m.communitiesManager.CreateAndSeedHistoryArchive(c.ID(), topics, lastArchiveEndDate, to, messageArchiveInterval)
+				if err != nil {
+					m.logger.Debug("failed to get create and seed history archive", zap.Error(err))
+				}
+
+				go m.communitiesManager.StartHistoryArchiveTasksInterval(c, messageArchiveInterval)
+			}
+		}
+	}
+}
+
+func (m *Messenger) dispatchMagnetlinkMessage(communityID string) error {
+
+	community, err := m.communitiesManager.GetByIDString(communityID)
+	if err != nil {
+		return err
+	}
+
+	magnetlink, err := m.communitiesManager.GetHistoryArchiveMagnetlink(community.ID())
+	if err != nil {
+		return err
+	}
+
+	magnetLinkMessage := &protobuf.CommunityMessageArchiveMagnetlink{
+		Clock:     m.getTimesource().GetCurrentTime(),
+		MagnetUri: magnetlink,
+	}
+
+	encodedMessage, err := proto.Marshal(magnetLinkMessage)
+	if err != nil {
+		return err
+	}
+
+	chatID := community.MagnetlinkMessageChannelID()
+	rawMessage := common.RawMessage{
+		LocalChatID:          chatID,
+		Sender:               community.PrivateKey(),
+		Payload:              encodedMessage,
+		MessageType:          protobuf.ApplicationMetadataMessage_COMMUNITY_ARCHIVE_MAGNETLINK,
+		SkipGroupMessageWrap: true,
+	}
+
+	log.Println("Dispatching magnet link for community ", community.IDString())
+	_, err = m.sender.SendPublic(context.Background(), chatID, rawMessage)
+	if err != nil {
+		return err
+	}
+
+	err = m.communitiesManager.UpdateCommunityDescriptionMagnetlinkMessageClock(community.ID(), magnetLinkMessage.Clock)
+	if err != nil {
+		return err
+	}
```

Comment:
We're updating the magnet link clock in the community description. This is needed so that, when users receive community invites, they get the latest magnetlink clock (otherwise, the moment they'll join, they'll try to process all past magnetlink messages they receive)